### PR TITLE
[Docs] Add concept pages for Resources and Blocks

### PR DESCRIPTION
**Pull request by movekevin**

### CODE REVIEW

**Review by clay-aptos — APPROVED** [review]
Thank, you Kevin! These LGTM with some minor formatting and edits I made in direct commits here. I will help get them into navigation and announced in See What's New next.

**Review by JoshLind — APPROVED** [review]
Unblocking.

File: developer-docs-site/docs/concepts/blocks.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## System transactions
+Each Aptos block contains both user transactions and special system transactions to *mark* the beginning and end of the transaction
+batch. Specifically, there are two system transactions:
+1. `BlockMetadataTransaction` - is inserted at the beginning of the block. A `BlockMetadata` transaction
+can also mark the end of an [epoch](#epoch) and trigger reward distribution to validators.
+2. `StateCheckpointTransaction` - is appended at the end of the block and is used as a checkpoint milestone.
+
+## Epoch
+In order to safely synchronize major changes such as validator set additions/removals, blocks on Aptos are further divided
+into epochs. Currently, a block is two hours long on Mainnet. During an epoch, blocks are produced yet major changes such
```
Comment (davidiw) [inline developer-docs-site/docs/concepts/blocks.md:26]: you folks really need to read this content :trollface:

Comment (clay-aptos) [reply developer-docs-site/docs/concepts/blocks.md:26]: > you folks really need to read this content :trollface:

Yes, please help review or find a suitable reviewer.

**Pull request merged**